File: LeetCode_Marathons/LCM6/Easy/Smallest_Range_I/a.py
class Solution:
    def smallestRangeI(self, nums: List[int], k: int) -> int:
        minNum: int = min(nums)
        maxNum: int = max(nums)

        if maxNum - minNum <= 2 * k: return 0
        return (maxNum - minNum) - 2 * k

File: LeetCode_Marathons/LCM6/Easy/Smallest_Range_I/a.java
class Solution {
    public int smallestRangeI(int[] nums, int k) {
        int minNum = Integer.MAX_VALUE, maxNum = Integer.MIN_VALUE;

        for (Integer num : nums) {
            if (num < minNum) minNum = num;
            if (num > maxNum) maxNum = num;
        }

        if (maxNum - minNum <= 2 * k) return 0;
        return maxNum - minNum - 2 * k;
    }
}